tests/features/security/xss_vulnerability.feature

```diff
@@ -32,3 +32,17 @@ Feature: Xss Vulnerability
     And I click "Article with xss Code in an image"
     And the response should contain "&lt;script&gt;alert('xss alert');&lt;/script&gt;"
     And the response should not contain "<script>alert('xss alert');</script>"
+
+   Scenario: Test the "Sanitize HTML" filter is never disabled if the "Display any HTML as plain text" one is in a "text format"
```

Comment:
If I understand correctly this scenario, you put two different scenarios in one scenario.
The first one, testing that one of the sanitize filters needs to be checked (and checks automatically the "sanitize html"), and the second one, checking that we can check the "Display any HTML as plain text" (and this one is not override).
So, may be, we can rewrite this behat test, in 3 separate behat tests:
- no sanitize filters checked, so after saving, the sanitize_html is checked and "Display any HTML as plain text" is not checked,
- only sanitize_html is checked, after saving, the sanitize_html is checked and "Display any HTML as plain text" is not checked,
- only "Display any HTML as plain text" is checked, after saving: the sanitize_html is not checked and "Display any HTML as plain text" is checked,

Furthermore, we can create a additional test with this user case, to understand better the scope of this ticket:
- create a new text format and check is "sanitize_html" is checked.

Reply:
I will add steps to cover the last point "create a new text format and check is "sanitize_html" is checked.", 

I will not follow you on splitting the scenario in 3 (4 with the last point) because the point here is what I put in the scenario description, no more no less and my test covers the different possibilities minus the 4th you mention and I will add right after this comment.